### DISCUSSION TITLE
Rich Presence: FormatType Percent

### DISCUSSION THREAD

**Discussion by ghost** [RetroAchievements/RAWeb#739] [Ideas]
It would be nice to being able to display a value in form of a **Percent** FormatType. Recent example would be the [Pole Position](http://retroachievements.org/game/11690) RP, the Game has a Max Speed of 200 MPH but the Game value only goes from 00 to A3 (HEX) / 163 (DEC). I could use the **Value** FormatType to create a 0-163 MPH RP but that wouldn't be accurate.

My suggestion is to add a **Percent** FormatType based on a Min to Max value, this would allow us to surpass the 100% mark and create 200% (and more) depending on the Max value. The **%** doesn't needs to be included here (it can be added separately in the RP), by doing so we can do more flexible and dynamic RPs when the ingame value doesn't represents the real Value/Percentage in the Game like Pole Position does with the 200 MPH.

The **Percent** FormatType could also be used for anything else from Story progression to Item collecting percentage.

**Comment by Jamiras** [comment]
If you know the upper limit (the value for 100%), you can use multiplication to simulate a percentage. The percentage value is "current value * 100 / max value", so as long as max value is a constant, you can merge it with the 100 to make a single multiplication (100/163 = 0.6135):
```
Speed(0xH1234 * 0.6135)
```
While this may be useful for progression or collecting, I wouldn't recommend it for something as volatile as speed. The Rich Presence value is only updated/sent to the server once every two minutes.
